webpack/webpack-embedded-dependencies-plugin/src/index.ts

```diff
@@ -0,0 +1,282 @@
+        PLUGIN_NAME,
+        (module, moduleCreateData: IWebpackModuleCreateData, resolveData) => {
+          const { resourceResolveData } = moduleCreateData;
+          const pkg: IPackageData | undefined = resourceResolveData?.descriptionFileData;
+          const filePath: string | undefined = resourceResolveData?.descriptionFileRoot;
+
+          if (pkg && filePath && filePath.includes('node_modules')) {
+            const key: PackageMapKey = makePackageMapKeyForPackage(pkg);
+            thirdPartyPackages.set(key, { dir: filePath, data: pkg });
+          }
+
+          return module;
+        }
+      );
+    });
+
+    // Tap into compilation so we can tap into compilation.hooks.processAssets
+    compiler.hooks.thisCompilation.tap(PLUGIN_NAME, (compilation) => {
+      compilation.hooks.processAssets.tapPromise(
+        { name: PLUGIN_NAME, stage: Compilation.PROCESS_ASSETS_STAGE_REPORT },
+        async (assets) => {
+          const rawPackages: FlattenedPackageEntry[] = Array.from(thirdPartyPackages).sort((first, next) => {
+            return first[0].localeCompare(next[0]);
+          });
+
+          const packages: IPackageData[] = [];
+
+          Async.forEachAsync(rawPackages, async ([, { dir, data }]) => {
+            const { name, version } = data;
+            let licenseSource: string | undefined;
+            const license: string | undefined = parseLicense(data);
+            const licensePath: string | undefined = await this._getLicenseFilePath(dir, compiler);
+            if (licensePath) {
+              licenseSource = this._licenseSourceCache.get(licensePath);
+              if (!licenseSource) {
+                licenseSource = await FileSystem.readFile(licensePath);
+                // Cache the license source content with the path as the identifier.
+                // The default license generator can use this later if it is enabled.
+                this._licenseSourceCache.set(licensePath, licenseSource);
+              }
+
+              const copyright: string | undefined =
+                (await this._parseCopyright(licenseSource)) || parsePackageAuthor(data);
+
+              packages.push({
+                name,
+                version,
+                license,
+                licensePath,
+                copyright
+              });
+            } else {
+              // If there is no license file path, we still should populate the other required fields
+              const copyright: string | undefined = parsePackageAuthor(data);
+
+              packages.push({
+                name,
+                version,
+                license,
+                copyright
+              });
+            }
+          }).catch((error: unknown) => {
+            this._emitWebpackError(compilation, 'Failed to process embedded dependencies', error);
+          });
+
+          const dataToStringify: IEmbeddedDependenciesFile = {
+            embeddedDependencies: packages
```

Comment:
Does the JSON file need the non-portable path to the license file, or would it be better to just inline the content of the license file?

Reply:
I think you are right, I was trying to overengineer memory usage here and keeping the format of the document slim and I think its best to just throw the source inside the embedded dep json file. `licensePath` is now removed since non-portable like you mentioned and now just have license source included.